dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/trackedentity/SecurityOwnershipValidator.java

```diff
@@ -145,15 +153,25 @@ private void checkOrgUnitInCaptureScope(
     }
   }
 
-  private void checkOrgUnitInSearchScope(
-      Reporter reporter, TrackerBundle bundle, TrackerDto dto, OrganisationUnit orgUnit) {
-    User user = bundle.getUser();
-
-    checkNotNull(user, USER_CANT_BE_NULL);
-    checkNotNull(orgUnit, ORGANISATION_UNIT_CANT_BE_NULL);
-
-    if (!organisationUnitService.isInUserSearchHierarchyCached(user, orgUnit)) {
-      reporter.addError(dto, ValidationCode.E1003, orgUnit, user);
+  private void checkWriteAccess(
+      Reporter reporter, TrackerDto dto, TrackedEntity te, TrackerBundle bundle) {
+    List<Program> programsToCheck =
+        programService.getAllPrograms().stream()
+            .filter(
+                p ->
+                    p.getTrackedEntityType() != null
+                        && p.getTrackedEntityType()
+                            .getUid()
+                            .equals(te.getTrackedEntityType().getUid()))
+            .toList();
+    if (!programsToCheck.isEmpty()
+        && programsToCheck.stream()
+            .noneMatch(
+                p ->
+                    trackerAccessManager
+                        .canWrite(UserDetails.fromUser(bundle.getUser()), te, p, false)
+                        .isEmpty())) {
+      reporter.addError(dto, ValidationCode.E1003, te.getUid());
     }
```

Comment:
a question about this. Should this be a general rule for checking access to write a te? Could this be moved to the ACL directly as we did for `canRead` of a te?

Reply:
This is an outdated version of the code.
I already moved it to the ACL